core/src/main/scala/neotypes/Driver.scala

```diff
@@ -2,89 +2,134 @@ package neotypes
+    streamingTransaction(config = TransactionConfig.default)
+
+  def streamingTransact[T](config: TransactionConfig)(txF: StreamingTransaction[S, F] => S[T]): S[T]
+
+  final def streamingTransact[T](txF: StreamingTransaction[S, F] => S[T]): S[T] =
+    streamingTransact(config = TransactionConfig.default)(txF)
+}
+
+object Driver {
+  private def txFinalizer[F[_]]: (Transaction[F], Option[Throwable]) => F[Unit] = {
+    case (tx, None)    => tx.commit
+    case (tx, Some(_)) => tx.rollback
+  }
+
+  private class DriverImpl[F[_]](driver: NeoDriver)
+                                (implicit F: Async[F]) extends Driver[F] {
+    override final def metrics: F[List[ConnectionPoolMetrics]] =
+      F.fromEither(
+        Try(driver.metrics).map(_.connectionPoolMetrics.asScala.toList).toEither
+      )
+
+    override final def transaction(config: TransactionConfig): F[Transaction[F]] =
+      F.async { cb =>
+        val (sessionConfig, transactionConfig) = config.getConfigs
+        val s = driver.asyncSession(sessionConfig)
+
+        s.beginTransactionAsync(transactionConfig).accept(cb) { tx =>
+          Right(Transaction[F](tx, s))
+        }
+      }
+
+    override final def transact[T](config: TransactionConfig)(txF: Transaction[F] => F[T]): F[T] =
+      transaction(config).guarantee(txF)(txFinalizer)
+
+    override final def close: F[Unit] =
+      F.async { cb =>
+        driver.closeAsync().acceptVoid(cb)
+      }
+  }
+
+  private final class StreamingDriverImpl[S[_], F[_]](driver: NeoDriver)
+                                                     (implicit S: Stream.Aux[S, F], F: Async[F])
+                                                     extends DriverImpl[F](driver) with StreamingDriver[S, F] {
+    override def streamingTransaction(config: TransactionConfig): S[StreamingTransaction[S, F]] = {
+      val (sessionConfig, transactionConfig) = config.getConfigs
+
+      val session = F.delay {
+        driver.rxSession(sessionConfig)
+      }
+
+      /** Use when https://github.com/neo4j/neo4j-java-driver/issues/797 is fixed.
+      S.fromF(session).flatMapS { s =>
+        s.beginTransaction(transactionConfig).toStream[S].mapS { tx =>
+          Transaction[S, F](tx, s)
+        }
+      }
+      */
+
+      // Workaround for neo4j-java-driver#797 -------------------------------------------
+      val streamingTx = session.flatMap { s =>
+        F.async[StreamingTransaction[S, F]] { cb =>
+          val rxs = s.asInstanceOf[org.neo4j.driver.internal.reactive.InternalRxSession]
+          val f = rxs.getClass.getDeclaredField("session")
+          f.setAccessible(true)
+          val ns = f.get(rxs).asInstanceOf[org.neo4j.driver.internal.async.NetworkSession]
+          ns.beginTransactionAsync(transactionConfig).accept(cb) { tx =>
+            val rxtx = new org.neo4j.driver.internal.reactive.InternalRxTransaction(tx)
+            Right(Transaction(rxtx, s))
+          }
+        }
+      }
+      S.fromF(streamingTx)
```

Comment:
I hate this, period.
But at least works... hope https://github.com/neo4j/neo4j-java-driver/issues/797 gets fixed soon.